Month3/LinkedList/LinkedList/TDLList.h

```diff
@@ -0,0 +1,404 @@
+#include <iostream>
+
+template<typename T>
+
+class TDLList
+{
+private:
+	struct FNode
+	{
+		T Element;
+		FNode* Next = nullptr;
+		FNode* Prev = nullptr;
+	};
+	FNode* Tail = nullptr;
+	FNode* Head = nullptr;
+	int Size = 0;
+
+public:
+	TDLList()
+	{
+
+	}
+	TDLList(T& NewElement)
+	{
+		Head = new FNode();
+		Head->Element = NewElement;
+		Tail = Head;
+		Size = 1;
+	}
+
+	TDLList(const TDLList<T>& OtherList)
+	{
+		FNode* Current = OtherList.Head;
+		for (int i = 0; i < OtherList.Size; i++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+	}
+
+	TDLList& operator=(const TDLList<T>& OtherList)
+	{
+		Clear();
+		FNode* Current = OtherList.Head;
+		for (int CurrentIndex = 0; CurrentIndex < OtherList.Size; CurrentIndex++)
+		{
+			AddTail(Current->Element);
+			Current = Current->Next;
+		}
+		return *this;
+	}
+
+	~TDLList()
+	{
+		Clear();
+	}
+
+
+	T& operator[](const int Index)
+	{
+		FNode* Current = Head;
+		if (Index < 0 || Index>Size)
+		{
+			return;
+		}
+		if (Index < (Size / 2))
+		{
+
+			for (int CurrentIndex = 0; CurrentIndex = Index; ++CurrentIndex)
+			{
+				Current = Current->Next;
+			}
+			return Current->Next;
+		}
+		else if (Index > (Size / 2))
+		{
+			for (int CurrentIndex = Size - 1; CurrentIndex = Index; --CurrentIndex)
+			{
+				Current = Current->Prev;
+			}
+			return Current->Prev;
+		}
+
+	}
+	const T& operator[](const int Index) const
+	{
+		FNode* Current = Head;
+		if (Index < 0 || Index>Size)
+		{
+			return;
+		}
+		if (Index < (Size / 2))
+		{
+
+			for (int CurrentIndex = 0; CurrentIndex = Index; ++CurrentIndex)
+			{
+				Current = Current->Next;
+			}
+			return Current->Next;
+		}
+		else if (Index > (Size / 2))
+		{
+			for (int CurrentIndex = Size - 1; CurrentIndex = Index; --CurrentIndex)
+			{
+				Current = Current->Prev;
+			}
+			return Current->Prev;
+		}
+	}
+
+
+	T& GetHead()
+	{
+		return Head->Element;
+	}
+	const T& GetHead() const
+	{
+		return Head->Element;
+	}
+
+	T& GetTail()
+	{
+		return Tail->Element;
+	}
+	const T& GetTail() const
+	{
+		return Tail->Element;
+	}
+
+	bool IsEmpty()
+	{
+		return Size == 0;
+	}
+
+	const int GetSize() const
```

Comment:
return types shouldn't be const unless you're returning a reference or a pointer